app/models/student_task.rb

```
@@ -1,50 +1,173 @@
+    @due_dates = args[:due_dates]
+  end
+
+  # create a new StudentTask instance from a Participant object.
+  def self.create_from_participant(participant)
+    assignment = participant.assignment
+    next_due   = DueDate.next_due_date(assignment)
+    # Fetch the real topic name from SignedUpTeam -> ProjectTopic
+    team       = participant.team
+    topic_name = SignedUpTeam.find_by(team_id: team&.id)&.project_topic&.topic_name
+    task = new(
+      assignment:         assignment.name,
```

RuboCop (1.87.0)

[convention] Align the keys of a hash literal if they span more than one line.

(Layout/HashAlignment)